src/prefect/core/task.py

```diff
@@ -375,9 +375,8 @@ def copy(self, **task_args: Any) -> "Task":
 
         new = copy.copy(self)
 
-        # ensure new slug is provided
-        if "slug" not in task_args:
-            task_args["slug"] = str(uuid.uuid4())
+        if new.slug and "slug" not in task_args:
+            task_args["slug"] = new.slug + "-copy"
```

Comment:
open to other suggestions here - this is super naive

Reply:
I opened #2806 to remove this entirely, so the naive approach isn't terrible now